packages/typegpu/src/data/vectorImpl.ts

```diff
@@ -0,0 +1,683 @@
+  get wzxy() { return new this._Vec4(this[3], this[2], this[0], this[1]); }
+  get wzxz() { return new this._Vec4(this[3], this[2], this[0], this[2]); }
+  get wzxw() { return new this._Vec4(this[3], this[2], this[0], this[3]); }
+  get wzyx() { return new this._Vec4(this[3], this[2], this[1], this[0]); }
+  get wzyy() { return new this._Vec4(this[3], this[2], this[1], this[1]); }
+  get wzyz() { return new this._Vec4(this[3], this[2], this[1], this[2]); }
+  get wzyw() { return new this._Vec4(this[3], this[2], this[1], this[3]); }
+  get wzzx() { return new this._Vec4(this[3], this[2], this[2], this[0]); }
+  get wzzy() { return new this._Vec4(this[3], this[2], this[2], this[1]); }
+  get wzzz() { return new this._Vec4(this[3], this[2], this[2], this[2]); }
+  get wzzw() { return new this._Vec4(this[3], this[2], this[2], this[3]); }
+  get wzwx() { return new this._Vec4(this[3], this[2], this[3], this[0]); }
+  get wzwy() { return new this._Vec4(this[3], this[2], this[3], this[1]); }
+  get wzwz() { return new this._Vec4(this[3], this[2], this[3], this[2]); }
+  get wzww() { return new this._Vec4(this[3], this[2], this[3], this[3]); }
+  get wwxx() { return new this._Vec4(this[3], this[3], this[0], this[0]); }
+  get wwxy() { return new this._Vec4(this[3], this[3], this[0], this[1]); }
+  get wwxz() { return new this._Vec4(this[3], this[3], this[0], this[2]); }
+  get wwxw() { return new this._Vec4(this[3], this[3], this[0], this[3]); }
+  get wwyx() { return new this._Vec4(this[3], this[3], this[1], this[0]); }
+  get wwyy() { return new this._Vec4(this[3], this[3], this[1], this[1]); }
+  get wwyz() { return new this._Vec4(this[3], this[3], this[1], this[2]); }
+  get wwyw() { return new this._Vec4(this[3], this[3], this[1], this[3]); }
+  get wwzx() { return new this._Vec4(this[3], this[3], this[2], this[0]); }
+  get wwzy() { return new this._Vec4(this[3], this[3], this[2], this[1]); }
+  get wwzz() { return new this._Vec4(this[3], this[3], this[2], this[2]); }
+  get wwzw() { return new this._Vec4(this[3], this[3], this[2], this[3]); }
+  get wwwx() { return new this._Vec4(this[3], this[3], this[3], this[0]); }
+  get wwwy() { return new this._Vec4(this[3], this[3], this[3], this[1]); }
+  get wwwz() { return new this._Vec4(this[3], this[3], this[3], this[2]); }
+  get wwww() { return new this._Vec4(this[3], this[3], this[3], this[3]); }
+}
+
+type Tuple2 = [number, number];
+type Tuple3 = [number, number, number];
+type Tuple4 = [number, number, number, number];
+
+abstract class Vec2 extends VecBase implements Tuple2 {
+  declare readonly length = 2;
+
+  0: number;
+  1: number;
+
+  constructor(x?: number, y?: number) {
+    super(2);
+    this[0] = x ?? 0;
+    this[1] = y ?? x ?? 0;
+  }
+
+  get x() {
+    return this[0];
+  }
+
+  get y() {
+    return this[1];
+  }
+
+  set x(value: number) {
+    this[0] = value;
+  }
+
+  set y(value: number) {
+    this[1] = value;
+  }
+}
+
+// biome-ignore format: swizzles should not expand
```

Comment:
Tiny nit, there are no swizzles in this class, what formatting is this comment ignoring?
(Also, the `Vec2` class does not have this comment)

Reply:
done